avm/res/api-management/service/api/diagnostics/main.bicep

```diff
@@ -0,0 +1,102 @@
+metadata name = 'API Management Service APIs Diagnostics.'
+metadata description = 'This module deploys an API Management Service API Diagnostics.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. The name of the parent API Management service.')
+param apiManagementServiceName string
+
+@description('Required. The name of the parent API.')
+param apiName string
+
+@description('Required. The name of the logger.')
+param loggerName string
+
+@allowed([
+  'azuremonitor'
+  'applicationinsights'
+  'local'
+])
+@description('Optional. Type of diagnostic resource. Default is local.')
+param diagnosticName string = 'local'
+
+@description('Optional. Specifies for what type of messages sampling settings should not apply.')
+param alwaysLog string = 'allErrors'
+
+@description('Optional. Diagnostic settings for incoming/outgoing HTTP messages to the Backend.')
+param backend object = {}
+
+@description('Optional. Diagnostic settings for incoming/outgoing HTTP messages to the Gateway.')
+param frontend object = {}
+
+@allowed([
+  'Legacy'
+  'None'
+  'W3C'
+])
+@description('Conditional. Sets correlation protocol to use for Application Insights diagnostics. Default is Legacy. Required if using Application Insights.')
+param httpCorrelationProtocol string = 'Legacy'
+
+@description('Optional. Log the ClientIP. Default is false.')
+param logClientIp bool = false
+
+@description('Conditional. Emit custom metrics via emit-metric policy. Default is false. Required if using Application Insights.')
+param metrics bool = false
+
+@allowed([
+  'Name'
+  'URI'
+])
+@description('Conditional. The format of the Operation Name for Application Insights telemetries. Default is Name. Required if using Application Insights.')
+param operationNameFormat string = 'Name'
+
+@description('Optional. Rate of sampling for fixed-rate sampling. Specifies the percentage of requests that are logged. 0% sampling means zero requests logged, while 100% sampling means all requests logged. Default is 100.')
+param samplingPercentage int = 100
+
+@allowed([
+  'error'
+  'information'
+  'verbose'
+])
+@description('Optional. The verbosity level applied to traces emitted by trace policies. Default is "error".')
+param verbosity string = 'error'
+
+resource service 'Microsoft.ApiManagement/service@2021-08-01' existing = {
+  name: apiManagementServiceName
+
+  resource api 'apis@2021-08-01' existing = {
+    name: apiName
+  }
+
+  resource logger 'loggers@2021-08-01' existing = {
+    name: loggerName
+  }
+}
+
+resource diagnostic 'Microsoft.ApiManagement/service/apis/diagnostics@2022-08-01' = {
+  name: diagnosticName
```

Comment:
Should not be called `diagnosticName` but name as per spec [Inputs - Parameter/Variable Naming](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-rmfr6---category-inputs---parametervariable-naming)